### PR TITLE
Bug 1763581: Increase OSD size from 1TiB to 2TiB

### PR DESCRIPTION
**Pull request by a2batic**
expand

Signed-off-by: Kanika <kmurarka@redhat.com>

### CODE REVIEW

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/_add-capacity-modal.scss [hunk 1]
```diff
@@ -4,6 +4,22 @@
...
+.add-capacity-modal__input {
+  margin-top: 15px;
+}
+
+.add-capacity-modal__input input {
+  max-width: 16rem;
+}
+
+.add-capacity-modal__input .dropdown {
+  margin-left: 1rem;
```
[the remark this answers is not in the document]
Comment (a2batic) [reply frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/_add-capacity-modal.scss]: ack

File: frontend/packages/ceph-storage-plugin/src/constants/ocs-install.ts [hunk 2]
```diff
@@ -31,7 +31,7 @@ export const ocsRequestData: K8sResourceKind = {
             volumeMode: 'Block',
             resources: {
               requests: {
-                storage: '1Ti',
+                storage: '2Ti',
```
[the remark this answers is not in the document]
Comment (a2batic) [reply frontend/packages/ceph-storage-plugin/src/constants/ocs-install.ts]: ack

File: frontend/public/components/utils/request-size-input.tsx [hunk 3]
```diff
@@ -59,4 +59,5 @@ export type RequestSizeInputProps = {
   defaultRequestSizeUnit: string;
   defaultRequestSizeValue: string;
   describedBy?: string;
+  step?: number;
```
Comment (afreen23) [inline frontend/public/components/utils/request-size-input.tsx]: Are these changes required, if we are disabling button.
We must avoid making changes to generic components of console

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx [hunk 4]
```diff
@@ -61,6 +62,8 @@ export const AddCapacityModal = withHandlePromise((props: AddCapacityModalProps)
 
   const handleRequestSizeInputChange = (capacityObj: any) => {
     setRequestSizeValue(capacityObj.value);
+    setValid(capacityObj.value % 2 === 0);
+    setButton(capacityObj.value % 2 !== 0);
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx]: One of them would be enough to check for validation.

File: frontend/public/components/utils/request-size-input.tsx [hunk 5]
```diff
@@ -27,7 +27,7 @@ export class RequestSizeInput extends React.Component<RequestSizeInputProps> {
           <input
             className="pf-c-form-control"
             type="number"
-            step="any"
+            step={this.props.step || 'any'}
```
Comment (afreen23) [inline frontend/public/components/utils/request-size-input.tsx]:
```suggestion
```

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx [hunk 6]
```diff
@@ -91,8 +94,12 @@ export const AddCapacityModal = withHandlePromise((props: AddCapacityModalProps)
               defaultRequestSizeUnit={requestSizeUnit}
               defaultRequestSizeValue={requestSizeValue}
               dropdownUnits={dropdownUnits}
+              step={2}
               required
             />
+            <p className={`text-secondary ${isValid ? '' : 'add-capacity-modal__help-text'}`}>
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx]:
```suggestion
            <p className={`text-secondary ${buttonDisabled ? '' : 'add-capacity-modal__help-text'}`}>
```

File: frontend/public/components/utils/request-size-input.tsx (hunk 3 above)
Comment (a2batic) [reply frontend/public/components/utils/request-size-input.tsx]: @afreen23, the OSD should be increased in factors of 2, so just to make a better user experience when up/down arrow keys are pressed, I have added `step` as an optional parameter. Since it can still have `any` as value, it is not affecting any other component where it is used. @spadgett what are your thoughts about this ?

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/_add-capacity-modal.scss [hunk 7]
```diff
@@ -4,6 +4,18 @@
   display: none;
 }
 
+.add-capacity-modal__input input {
+  max-width: 16rem;
```
Comment (gnehapk) [inline frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/_add-capacity-modal.scss]: I feel, px should be preferred here as rem depends on root's node size and it can be changed by browser settings.

Comment (gnehapk) [reply frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/_add-capacity-modal.scss]: also targeting html tag direct should be avoided as later on, if any input tag gets added, it can break the style.

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx [hunk 8]
```diff
@@ -2,8 +2,8 @@ import * as React from 'react';
 import * as _ from 'lodash';
 import {
   FieldLevelHelp,
-  RequestSizeInput,
   withHandlePromise,
+  RequestSizeInput,
```
Comment (gnehapk) [inline frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx]: Alpha imports

**Comment by a2batic** [comment]
@rawagner @spadgett  please review.

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx [hunk 9]
```diff
@@ -91,8 +92,18 @@ export const AddCapacityModal = withHandlePromise((props: AddCapacityModalProps)
               defaultRequestSizeUnit={requestSizeUnit}
               defaultRequestSizeValue={requestSizeValue}
               dropdownUnits={dropdownUnits}
+              step={2}
+              minValue={2}
+              inputClassName="add-capacity-modal__input--width"
               required
             />
+            <p
+              className={`text-secondary add-capacity-modal__help-text ${
```
Comment (rawagner) [inline frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx]: you can use `classNames` here

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx [hunk 10]
```diff
@@ -73,14 +74,14 @@ export const AddCapacityModal = withHandlePromise((props: AddCapacityModalProps)
...
             <label className="control-label" htmlFor="request-size-input">
-              Requested Capacity
+              Capacity
               <FieldLevelHelp>{labelTooltip}</FieldLevelHelp>
               <span className="add-capacity-modal__span">
                 <span>
-                  Provisioned Capacity:
-                  {presentCount ? ` ${presentCount / 3}Ti` : ' Unavailable'}
+                  Provisioned:
+                  {presentCount ? ` ${presentCount * 2}TiB` : ' Unavailable'}
```
Comment (rawagner) [inline frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx]: shouldn't `Unavailable` be `text-secondary` ?

File: frontend/public/components/utils/request-size-input.tsx [hunk 11]
```diff
@@ -25,15 +25,16 @@ export class RequestSizeInput extends React.Component<RequestSizeInputProps> {
       <div className="form-group">
         <div className="pf-c-input-group">
           <input
-            className="pf-c-form-control"
+            className={`${this.props.inputClassName || ''} pf-c-form-control`}
```
Comment (rawagner) [inline frontend/public/components/utils/request-size-input.tsx]: you can use `classNames` here too

**Comment by afreen23** [comment]
FYI, the validations should come from backend and the submit button should be left enabled as per the **console convention**

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx (hunk 9 above)
Comment (a2batic) [reply frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx]: ack

File: frontend/public/components/utils/request-size-input.tsx (hunk 11 above)
Comment (a2batic) [reply frontend/public/components/utils/request-size-input.tsx]: ack

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx (hunk 10 above)
Comment (a2batic) [reply frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx]: ack

File: frontend/public/components/utils/request-size-input.tsx (hunk 3 above)
Comment (spadgett) [reply frontend/public/components/utils/request-size-input.tsx]: This should be fine since the behavior is unchanged if I don't provide a `step` prop
